.github/actions/npm-blacklist/checker.sh

```
@@ -0,0 +1,113 @@
+
+# ---
+
+# Track if any matches were found
+found_matches=false
+alerts=() # Array to store alert messages
+
+echo "🔍 Checking for blacklisted packages in '$project_path' dependency tree..."
+echo ""
+
+# Run npm ls in the specified directory and get the full dependency tree
+dep_tree=$(cd "$project_path" && npm ls --all --silent 2>/dev/null || true)
```

Shellcheck (0.10.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)